tuf/client/updater.py

```diff
@@ -3237,6 +3253,22 @@ def download_target(self, target, destination_directory,
         to ensure snapshot consistency).
         Default is True.
 
+     custom_download_handler:
+        A user provided function performing the actual target file download.
+        In order to comply with the TUF specification, the function signature
+        should have the form:
+
+        'def download_handler_func(url, required_length)'
```

Comment:
Can we encode the expected API of the handler somehow and verify that a passed `custom_download_handler` adheres to it? 

What must be included in that API? Certainly the number, order and types of the arguments the handler expects. What about exceptions? The `download_target()` call stack can potentially throw quite a few, for example `_download_file` throws a `tuf.exceptions.DownloadLengthMismatchError`

I'd very much welcome some input here from users of Updater – @trishankatdatadog @jku any thoughts?

Reply:
We can use [inspect.signature](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-objec) for that purpose. Unfortunately the doc says _New in version 3.3._

Reply:
As long as it does not affect current users (can upgrade transparently w/o changing anything or seeing any changes), I have no strong opinion 🙂